unittest/unit/update-bcd.ts

```diff
@@ -983,6 +1027,290 @@ describe('BCD updater', () => {
         },
         userAgent: firefox92UaString
       };
+
+      const sm = getSupportMatrix([report], initialBcd.browsers, []);
+
+      const modified = update(finalBcd, sm, {});
+
+      assert.equal(modified, false, 'modified');
+      assert.deepEqual(finalBcd, initialBcd);
+    });
+
+    it('skips negative results for unsupported features', () => {
+      const initialBcd = {
+        api: {
+          AbortController: {
+            __compat: {
+              support: {
+                firefox: {version_added: '91', flags: [{}]}
+              }
+            }
+          }
+        },
+        browsers: {
+          firefox: {name: 'Firefox', releases: {91: {}, 92: {}, 93: {}}}
+        } as unknown as Browsers
+      };
+      const finalBcd = clone(initialBcd);
+      const report: Report = {
+        __version: '0.3.1',
+        results: {
+          'https://mdn-bcd-collector.appspot.com/tests/': [
+            {
+              name: 'api.AbortController',
+              exposure: 'Window',
+              result: false
+            }
+          ]
+        },
+        userAgent: firefox92UaString
+      };
+
+      const sm = getSupportMatrix([report], initialBcd.browsers, []);
+
+      const modified = update(finalBcd, sm, {});
+
+      assert.equal(modified, false, 'modified');
+      assert.deepEqual(finalBcd, initialBcd);
+    });
+
+    it('skips complex support scenarios', () => {
```

Comment:
This is a bit tricky: I think the situation you're describing is inaccurate, but your concern is valid and does indeed warrant a new test--just not the test you've suggested.

The test titled "skips removed features" exercises the branch you've highlighted. That test ought to fail when the corresponding TODO comment is addressed.

On the other hand, the test named "skips complex support scenarios" hits the branch immediately before the one you've highlighted:

https://github.com/foolip/mdn-bcd-collector/blob/19a623062ab0a82da8d3cc7c757452a5e408f877/update-bcd.ts#L381-L384

(The "TODO" comment there inspired the name of this test.) You're right that this will continue to pass after the "TODO" comment is addressed, so I agree that I should add another test which would fail when the skipping is replaced with more advanced logic. That test needs two "default statements" (to use the terminology of the source code), so I've selected a "support" value of: `[{version_added: '93'}, {version_removed: '94'}]`. When the "update" script is extended to handle this situation, it ought to change `version_added`, and the new test ought to fail.